plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/expression/SparkExpressionBuilder.java

```diff
@@ -0,0 +1,231 @@
+    {
+        // Handle comparison operator
+        if (context.predicate() != null) {
+            return visit(context.predicate());
+        }
+
+        // Handle simple expression likes just TRUE
+        return visit(context.valueExpression);
+    }
+
+    @Override
+    public Object visitComparison(SparkExpressionParser.ComparisonContext context)
+    {
+        return new ComparisonExpression(
+                getComparisonOperator(((TerminalNode) context.comparisonOperator().getChild(0)).getSymbol()),
+                (SparkExpression) visit(context.value),
+                (SparkExpression) visit(context.right));
+    }
+
+    @Override
+    public SparkExpression visitAnd(SparkExpressionParser.AndContext context)
+    {
+        verify(context.booleanExpression().size() == 2, "AND operator expects two expressions: " + context.booleanExpression());
+        return new LogicalExpression(LogicalExpression.Operator.AND, visit(context.booleanExpression(0), SparkExpression.class), visit(context.booleanExpression(1), SparkExpression.class));
+    }
+
+    @Override
+    public Object visitColumnReference(SparkExpressionParser.ColumnReferenceContext context)
+    {
+        return visit(context.identifier());
+    }
+
+    private static ComparisonExpression.Operator getComparisonOperator(Token symbol)
+    {
+        switch (symbol.getType()) {
+            case SparkExpressionLexer.EQ:
+                return ComparisonExpression.Operator.EQUAL;
+            case SparkExpressionLexer.NEQ:
+                return ComparisonExpression.Operator.NOT_EQUAL;
+            case SparkExpressionLexer.LT:
+                return ComparisonExpression.Operator.LESS_THAN;
+            case SparkExpressionLexer.LTE:
+                return ComparisonExpression.Operator.LESS_THAN_OR_EQUAL;
+            case SparkExpressionLexer.GT:
+                return ComparisonExpression.Operator.GREATER_THAN;
+            case SparkExpressionLexer.GTE:
+                return ComparisonExpression.Operator.GREATER_THAN_OR_EQUAL;
+        }
+
+        throw new IllegalArgumentException("Unsupported operator: " + symbol.getText());
+    }
+
+    @Override
+    public Object visitBooleanLiteral(SparkExpressionParser.BooleanLiteralContext context)
+    {
+        return new BooleanLiteral(context.getText());
+    }
+
+    @Override
+    public SparkExpression visitIntegerLiteral(SparkExpressionParser.IntegerLiteralContext context)
+    {
+        return new LongLiteral(context.getText());
+    }
+
+    @Override
+    public Object visitUnicodeStringLiteral(SparkExpressionParser.UnicodeStringLiteralContext context)
+    {
+        return new StringLiteral(decodeUnicodeLiteral(context));
+    }
+
+    private static String decodeUnicodeLiteral(SparkExpressionParser.UnicodeStringLiteralContext context)
+    {
+        String rawContent = unquote(context.getText());
+        StringBuilder unicodeStringBuilder = new StringBuilder();
```

Comment:
every Java string is "a unicode string".
we're just parsing a string literal, so let's call it `result` or `value`